### PR TITLE
Upgrading to doc v3

### PR DESCRIPTION
**Pull request by npiyush97**
## Hey, I just made a Pull Request!
related to #23910

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by npiyush97** [comment]
@awanlin I don't understand why ci failing prettier:check? if there is prettier issue microsite pull should also fail. because of different versions?

**Comment by awanlin** [comment]
Hi @npiyush97 CI looks fine to me, just has that CodeQL error to sort out.

File: microsite/package.json
Comment (awanlin) [inline microsite/package.json:1]: We use `.mdx` files in the Blog and I think we might want to use them in the docs as well. We don't have the `"@mdx-js/react": "^3.0.0"` dependency in here at all, can we add that? And then anything else we might need from this: https://docusaurus.io/docs/migration/v3#upgrading-mdx

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for running with this @npiyush97, just left a comment related to MDX

**Comment by npiyush97** [comment]
@Rugvip ready for review

File: microsite/package.json
Comment (awanlin) [reply microsite/package.json:1]: @npiyush97 not sure if this got missed?

**Comment by npiyush97** [comment]
@awanlin https://www.npmjs.com/package/@mdx-js/react#when-should-i-use-this
Its a Context provides a way to pass data through the component tree without having to pass props down manually at every level.

File: microsite/package.json
Comment (awanlin) [reply microsite/package.json:1]: But it's clearly a dependency in the Docusaurus upgrade documentation and that goes against this line here: https://docusaurus.io/docs/migration/v3#upgrading-mdx:~:text=MDX%20is%20a%20major%20dependency%20of%20Docusaurus%20responsible%20for%20compiling%20your%20.md%20and%20.mdx%20files%20to%20React%20components.

**Comment by npiyush97** [comment]
@benjdlambert for this 
https://github.com/backstage/backstage/blob/91a7e3294a4530b63717458facfb91f7ea883e44/microsite/docusaurus.config.ts#L75

**Review by awanlin — APPROVED** [review]
Thanks for getting this completed @npiyush97 🚀

**Comment by npiyush97** [comment]
@Rugvip if pr is alright can you merge this conflict resolve.

**Comment by npiyush97** [comment]
closing too many conflict move to #24294 24294

**Pull request closed without merging**